File: Parallel-Java/src/parallel/chrono/DateTimeGreater.java
package parallel.chrono;

import java.time.LocalDateTime;
public class DateTimeGreater {
    //reuse
    // True if time1 is after time2
    public boolean timeGreater(LocalDateTime time1, LocalDateTime time2){
        return time1.isAfter(time2);
    }

    public boolean yearGreater(LocalDateTime time1, LocalDateTime time2){
        return time1.getYear() > time2.getYear();
    }

    public boolean monthGreater(LocalDateTime time1, LocalDateTime time2){
        return time1.getMonth().getValue() > time2.getMonth().getValue();
    }

    public boolean dayOfMonthGreater(LocalDateTime time1, LocalDateTime time2){
        return time1.getDayOfMonth() > time2.getDayOfMonth();
    }
    // Monday is 1 and Sunday is 7
    public boolean dayOfWeekGreater(LocalDateTime time1, LocalDateTime time2){
        return time1.getDayOfWeek().getValue() > time2.getDayOfWeek().getValue();
    }

    public boolean hourGreater(LocalDateTime time1, LocalDateTime time2){
        return time1.getHour() > time2.getHour();
    }

    public boolean minuteGreater(LocalDateTime time1, LocalDateTime time2){
        return time1.getMinute() > time2.getMinute();
    }

    public boolean secondGreater(LocalDateTime time1, LocalDateTime time2){
        return time1.getSecond() > time2.getSecond();
    }
    // reuse end

}

File: Parallel-Csharp/src/Parallel/Chrono/DateTimeGreater.cs
namespace Parallel.Linq101.Chrono;
using System;
public class DateTimeGreater {
    //reuse
    public bool TimeGreater(DateTime time1, DateTime time2){
        return time1 > time2;
    }

    public bool YearGreater(DateTime time1, DateTime time2){
        return time1.Year > time2.Year;
    }

    public bool MonthGreater(DateTime time1, DateTime time2){
        return time1.Month > time2.Month;
    }

    public bool DayOfMonthGreater(DateTime time1, DateTime time2){
        return time1.Day > time2.Day;
    }
    
    //Sunday is 0, while in Java Sunday is 7
    //We follow Java routine, convert Sunday to the highest order.
    public bool DayOfWeekGreater(DateTime time1, DateTime time2)
    {
        int value1 = (int)time1.DayOfWeek;
        int value2 = (int)time2.DayOfWeek;
        if (value1 == 0) value1 += 7;
        if (value2 == 0) value2 += 7;
        return value1 > value2;
    }

    public bool HourGreater(DateTime time1, DateTime time2){
        return time1.Hour > time2.Hour;
    }

    public bool MinuteGreater(DateTime time1, DateTime time2){
        return time1.Minute > time2.Minute;
    }

    public bool SecondGreater(DateTime time1, DateTime time2){
        return time1.Second > time2.Second;
    }
    // reuse end
}